src/lib/storage/constraints/unique_checker.cpp

```diff
@@ -0,0 +1,63 @@
+#include <string>
+
+#include "storage/constraints/unique_checker.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace opossum {
+
+
+bool constraint_valid_for(const Table& table, const TableConstraintDefinition& constraint,
+                          const CommitID& snapshot_commit_id, const TransactionID& our_tid) {
+  std::set<boost::container::small_vector<AllTypeVariant, 3>> unique_values;
```

Comment:
Was `std::set` the fastest here? We might have had this discussion before, but did you try it with `unordered_set` and `(vector.sort().unique()).size() == vector.size()`?

Reply:
We're going to use a sorted vector in the optimized next version.